src/modules/map/components/toolbox/TimeSlider.vue

```diff
@@ -273,30 +233,52 @@ function releaseCursor() {
     window.removeEventListener('touchend', releaseCursor)
     store.dispatch('setPreviewYear', { year: currentYear.value, ...dispatcher })
 }
+
 function togglePlayYearsWithData() {
     playYearsWithData = !playYearsWithData
     if (playYearsWithData) {
-        // if current year is the last (most recent) one, we set the starting year for our
-        // player to the oldest
-        if (currentYear.value === yearsWithData.value[0]) {
-            setCurrentYearAndDispatchToStore(yearsWithData.value.slice(-1)[0])
+        let yearsWithDataForPlayer = ALL_YEARS.filter(
+            (year) =>
+                yearsWithData.value.yearsJoint.includes(year) ||
+                yearsWithData.value.yearsSeparate.includes(year)
+        )
+            .sort()
+            .reverse()
+        // if current year is the last (most recent) one, or we're not on a year with data, we set the starting year for our
+        // player to the oldest year with data
+        if (
+            !yearsWithDataForPlayer.includes(currentYear.value) ||
+            currentYear.value === yearsWithDataForPlayer[0]
+        ) {
+            setCurrentYearAndDispatchToStore(yearsWithDataForPlayer.slice(-1)[0])
         }
         playYearInterval = setInterval(() => {
-            const currentYearIndex = yearsWithData.value.indexOf(currentYear.value)
+            const currentYearIndex = yearsWithDataForPlayer.indexOf(currentYear.value)
             // if last (most recent) year, we stop the player
             if (currentYearIndex === 0) {
                 clearInterval(playYearInterval)
                 playYearInterval = null
                 playYearsWithData = false
             } else {
-                setCurrentYearAndDispatchToStore(yearsWithData.value[currentYearIndex - 1])
+                setCurrentYearAndDispatchToStore(yearsWithDataForPlayer[currentYearIndex - 1])
             }
         }, 1000)
     } else {
         clearInterval(playYearInterval)
         playYearInterval = null
     }
 }
+
+function timeSliderBarInnerStepClassObject(year, index) {
+    const classObject = {
+        'has-joint-data': yearsWithData.value.yearsJoint.includes(year),
+        'big-tick': year % 50 === 0,
+        'medium-tick': year % 25 === 0,
+        'small-tick': year % 5 === 0,
+    }
+    classObject[`has-separate-data-${index % 2}`] = yearsWithData.value.yearsSeparate.includes(year)
+    return classObject
+}
```

Comment:
I'd prefer we keep CSS/HTML classes defined in the HTML template if possible
Here I see that you moved that here to be able to use the `index % 2` inside the function, maybe switch to a something like
```
{
// ...
'separate-data': yearsWithData.yearsSeparate.includes(year),
'odd': index % 2
}
```
and then in the (S)CSS
```
.separate-data {
    background: // ...
    &.odd {
       background: // ... will override above rule
    }
}
```